server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java

```diff
@@ -319,46 +322,47 @@ default void dequeueShouldAllowRetrieveFailItems() throws Exception {
             .name("name1")
             .build());
 
-        MailQueue.MailQueueItem mailQueueItem1 = getMailQueue().deQueue();
+        Iterator<MailQueue.MailQueueItem> items = getMailQueue().deQueue().subscribeOn(Schedulers.elastic()).toIterable().iterator();
+        MailQueue.MailQueueItem mailQueueItem1 = items.next();
         mailQueueItem1.done(false);
-        MailQueue.MailQueueItem mailQueueItem2 = getMailQueue().deQueue();
+        MailQueue.MailQueueItem mailQueueItem2 = items.next();
         mailQueueItem2.done(true);
         assertThat(mailQueueItem1.getMail().getName()).isEqualTo("name1");
         assertThat(mailQueueItem2.getMail().getName()).isEqualTo("name1");
     }
 
     @Test
-    default void dequeueShouldNotReturnInProcessingEmails(ExecutorService executorService) throws Exception {
+    default void dequeueShouldNotReturnInProcessingEmails() throws Exception {
         enQueue(defaultMail()
             .name("name")
             .build());
 
-        getMailQueue().deQueue();
+        LinkedBlockingQueue<MailQueue.MailQueueItem> q = new LinkedBlockingQueue<>(1);
```

Comment:
Can you add a better name for `q` (ex.: `queue`)

Reply:
ok